xml_converter/generators/code_generator.py

```diff
@@ -368,26 +367,53 @@ def write_attribute (self, output_directory: str) -> None:
         categories: Dict[str,List[str]] = {}
         attribute_names: Dict[str,str] = {}
         metadata: Dict[str, SchemaType] = {}
-         
+        attribute_variables: Tuple[List[Tuple[str, str]]] = []
+        doc_type_to_cpp_type: Dict[str,str] = {
+            "Fixed32": "int",
+            "Int32": "int",
+            "Boolean": "bool",
+            "Float32": "float",
+            "String": "string",
+        }
         
         for filepath in self.data.keys():
             filename = os.path.basename(filepath)
             attribute_names[filepath]= filename.split(".md")[0]
 
         for filepath in attribute_names:
+            attribute_variables = []
             attribute_name = attribute_names[filepath]
             metadata[filepath] = self.data[filepath].metadata
-            if metadata[filepath]['type'] == ["MultiflagValue"]:
-                attribute_variables = metadata[filepath]['flags']
-                class_name = capitalize(attribute_name,delimiter="") 
+            if metadata[filepath]['type'] in ["MultiflagValue", "CompoundValue", "Enum"]:
+                if metadata[filepath]['type'] == "MultiflagValue":
+                    for flag in metadata[filepath]['flags']:
+                        attribute_variables.append((flag, "bool")) 
+                        
+                elif metadata[filepath]['type'] == "CompoundValue":
+                    
+                    for component in metadata[filepath]['components']:
+                        if component['type'] not in doc_type_to_cpp_type:
+                            raise ValueError("Unexpected type for component. Look at markdown file {attribute_name}".format(
+                                attribute_name=attribute_name
+                            ))
+                        
+                        attribute_variables.append((component['name'].lower().replace(" ","_"), doc_type_to_cpp_type[component['type']]))
```

Comment:
Maybe think about making a `lowercase` function to pair with the `capitalize` function below.

Reply:
Done